XSConsoleData.py

```diff
@@ -1039,7 +1039,22 @@ def ReconfigureManagement(self, inPIF, inMode,  inIP,  inNetmask,  inGateway, in
         Auth.Inst().AssertAuthenticated()
         try:
             self.RequireSession()
-            self.session.xenapi.PIF.reconfigure_ip(inPIF['opaqueref'],  inMode,  inIP,  inNetmask,  inGateway, FirstValue(inDNS, ''))
+            if inPIF['primary_address_type'].lower() == 'ipv4':
+                self.session.xenapi.PIF.reconfigure_ip(inPIF['opaqueref'],  inMode,  inIP,  inNetmask,  inGateway, FirstValue(inDNS, ''))
+                if inPIF['ipv6_configuration_mode'].lower() == 'static':
```

Comment:
The expressions in the ifs can raise exceptions when the keys are not defined, please add presence checks before checking the contents